### ISSUE TITLE
Build Agent 2.105.7 becomes idle and does not respond

### ISSUE THREAD

**Issue by GrimSmiler** [Microsoft/vsts-agent#928]
## **Agent version and platform**
2.105.7
<br>
Server 2012 R2 x64
<br>
## **VSTS type and version**
On-Prem TFS
<br>
TFS 2017	- RTW - 15.105.25910.0
<br>
## **What's not working?** 
Sometimes after the agent gets a cancellation of a build, he crashes and becomes idle.
New builds will not get started and will show a message  "Waiting for the build agent", while the build agent will show "Idle"

Only work around - kill all instances of the agent, remove it from the agent pool and reconfigure

Agent is not running as a service

## **Agent and Worker's diag log**
If the crash happens, it is possible to observe the following error in the Agent_ log file
```
[2017-04-19 09:16:56Z INFO JobDispatcher] Ensure WorkerDispather for job 0835b619-12cb-4477-b393-2b179f45c353 run to finish, cancel any running job.
[2017-04-19 09:16:56Z ERR  JobDispatcher] Worker Dispatch failed witn an exception for job request 0835b619-12cb-4477-b393-2b179f45c353.
[2017-04-19 09:16:56Z ERR  JobDispatcher] System.IO.IOException: The handle is invalid
   at System.ConsolePal.WindowsConsoleStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.SyncTextWriter.WriteLine(String value)
   at System.Console.WriteLine(String value)
   at Microsoft.VisualStudio.Services.Agent.Terminal.WriteLine(String line)
   at Microsoft.VisualStudio.Services.Agent.Listener.JobDispatcher.<RunAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.JobDispatcher.<EnsureDispatchFinished>d__10.MoveNext()
[2017-04-19 09:16:56Z ERR  Terminal] WRITE ERROR: An error occurred: The handle is invalid
[2017-04-19 09:16:56Z ERR  AgentProcess] System.IO.IOException: The handle is invalid
   at System.ConsolePal.WindowsConsoleStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.SyncTextWriter.WriteLine(String value)
   at System.Console.WriteLine(String value)
   at Microsoft.VisualStudio.Services.Agent.Terminal.WriteLine(String line)
   at Microsoft.VisualStudio.Services.Agent.Listener.MessageListener.<CreateSessionAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.MessageListener.<GetNextMessageAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.Agent.<RunAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Services.Agent.Listener.Agent.<RunAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Services.Agent.Listener.Agent.<RunAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.Agent.<ExecuteCommand>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.Program.<MainAsync>d__1.MoveNext()
```
Thank You in advance. Sorry if this is a duplicate, but the crash happens now and than for almost a month now and i haven't found a solution. Please, if any log files are needed, specify which ones.

**Comment by bryanmacfarlane** [comment]
Have you tried a newer agent?  You can download 115

**Comment by GrimSmiler** [comment]
Good point. Will try it out right now. I'll notify You about the outcome.

Thanks for the hint!

**Comment by TingluoHuang** [comment]
this is a known issue, fix was checked in around version 2.109.x.
reopen the issue if latest agent doesn't solve it.
